File: packages/shared/src/types/conversation.ts
export type Channel = "phone" | "web_chat" | "sms";

export type MessageRole = "user" | "assistant" | "system";

export interface Message {
  role: MessageRole;
  content: string;
  timestamp: string;
}

export interface Conversation {
  id: string;
  tenant_id: string;
  lead_id: string | null;
  channel: Channel;
  transcript: Message[];
  summary: string | null;
  created_at: string;
  updated_at: string;
}

export interface ChatRequest {
  message: string;
  conversation_id?: string;
}

export interface ChatStreamEvent {
  type: "message" | "done" | "error";
  content?: string;
  error?: string;
}

File: apps/api/app/schemas/conversation.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.conversation import Channel


class MessageCreate(BaseModel):
    content: str


class ConversationResponse(BaseModel):
    id: uuid.UUID
    tenant_id: str
    lead_id: uuid.UUID | None
    channel: Channel
    transcript: list[dict]
    summary: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PaginatedConversationResponse(BaseModel):
    items: list[ConversationResponse]
    total_count: int
